catalog/video/video-admin-plugin/src/main/java/org/codice/alliance/video/ui/service/StreamMonitorHelper.java

```diff
@@ -0,0 +1,219 @@
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.framework.ServiceReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Metacard;
+
+public class StreamMonitorHelper implements StreamMonitorHelperMBean {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StreamMonitorHelper.class);
+
+    private static final String MONITORING = "monitoring";
+
+    private static final String SERVICE_PID = "service.Pid";
+
+    private static final String START_TIME = "startTime";
+
+    private ObjectName objectName;
+
+    private MBeanServer mBeanServer;
+
+    private BundleContext bundleContext;
+
+    public StreamMonitorHelper(BundleContext bundleContext) {
+        this.bundleContext = bundleContext;
+    }
+
+    public StreamMonitorHelper() {
+
+    }
+
+    @Override
+    public void callStartMonitoringStreamByServicePid(String servicePid) {
+        Map<String, org.codice.alliance.video.stream.mpegts.StreamMonitor> udpStreamMonitors =
+                getUdpStreamMonitorServices();
+
+        if (MapUtils.isEmpty(udpStreamMonitors)) {
+            return;
+        }
+
+        udpStreamMonitors.entrySet()
+                .stream()
+                .filter(entry -> entry.getKey()
+                        .equals(servicePid))
+                .forEach(stringStreamMonitorEntry -> {
+                    stringStreamMonitorEntry.getValue()
+                            .startMonitoring();
+                    LOGGER.debug("Started monitor for {}", stringStreamMonitorEntry.getKey());
+                });
+    }
+
+    @Override
+    public void callStopMonitoringStreamByServicePid(String servicePid) {
+        Map<String, org.codice.alliance.video.stream.mpegts.StreamMonitor> udpStreamMonitors =
+                getUdpStreamMonitorServices();
+
+        if (MapUtils.isEmpty(udpStreamMonitors)) {
+            return;
+        }
+
+        udpStreamMonitors.entrySet()
+                .stream()
+                .filter(entry -> entry.getKey()
+                        .equals(servicePid))
+                .forEach(stringStreamMonitorEntry -> {
+                    stringStreamMonitorEntry.getValue()
+                            .stopMonitoring();
+                    LOGGER.debug("Stopped monitor for {}", stringStreamMonitorEntry.getKey());
+                });
+    }
+
+    @Override
+    public List<Map<String, Object>> udpStreamMonitors() {
+        Map<String, org.codice.alliance.video.stream.mpegts.StreamMonitor> udpStreamMonitors =
+                getUdpStreamMonitorServices();
+
+        if (MapUtils.isEmpty(udpStreamMonitors)) {
+            return null;
+        }
+
+        List<Map<String, Object>> monitorList = new ArrayList<>();
+        udpStreamMonitors.entrySet()
+                .stream()
+                .forEach(stringStreamMonitorEntry -> {
```

Comment:
This might make your intent clearer:
```
return udpStreamMonitors.entrySet()
    .stream()
    .filter(entry -> entry.getValue() instanceof UdpStreamMonitor)
    .map(entry -> {
        ...
        return map;
    })
    .collect(Collectors.toList());
```